core/src/main/scala/org/apache/spark/deploy/client/StandaloneAppClient.scala

```diff
@@ -321,4 +321,15 @@ private[spark] class StandaloneAppClient(
     }
   }
 
+  /**
+   * Notify the Master about the removal of the executor
+   */
+  def executorRemoved(executorId: String): Unit = {
+    if (endpoint.get != null && appId.get != null) {
+      endpoint.get.send(ExecutorRemoved(appId.get, executorId))
```

Comment:
maybe avoid calling `endpoint.get` and `appId.get` repeatedly?

Reply:
This follows above `killExecutors` & `requestTotalExecutors`. Do you think we could improve all of them?